batch/test/test_batch.py

```diff
@@ -134,7 +134,15 @@ def test_deleted_job_log(self):
         id = j.id
         j.wait()
         j.delete()
-        self.assertEqual(self.batch._get_job_log(id), {'main': 'test\n'})
+
+        try:
+            self.batch._get_job_log(id)
+        except requests.HTTPError as e:
+            if e.response.status_code == 404:
+                pass
+            else:
+                raise
```

Comment:
`self.assertTrue(False, f"batch should not have deleted log {e}")` is preferable to indicate test failure